design/ceph/resource-dependencies.md

```diff
@@ -0,0 +1,259 @@
+  - OR
+  - `spec.storageClassName == <name of StorageClass which references the CephBlockPool>`
+
+Dependents via FlexVolume:
+- Unnecessary because Rook plans to deprecate FlexVolume support for v1.7
+
+#### `CephFilesystem`
+Dependents which can consume this provider's pools:
+- CephNFS in the same namespace that have `spec.pool == <a provider pool>`
+- CephClient in the same namespace that has any `spec.caps` value with the string `pool=<a provider pool>`
+
+Dependents via CSI:
+- StorageClasses that have:
+  - `provisioner == <operator namespace>.cephfs.csi.ceph.com` AND
+  - `parameters.clusterID == <ceph cluster namespace>` AND
+  - `parameters.pool` OR `parameters.dataPool` references a `CephBlockPool` pool
+- PersistentVolumes that have:
+  - `spec.CSI.Driver` == `<operator namespace>.cephfs.csi.ceph.com` AND
+  - `spec.CSI.VolumeAttributes["clusterID"] == <ceph cluster namespace>` AND
+  - `spec.CSI.VolumeAttributes["pool"]` OR `spec.CSI.VolumeAttributes["journalPool"]` references a `CephBlockPool` pool
+  - OR
+  - `spec.storageClassName == <name of StorageClass which references the CephBlockPool>`
+
+Dependents via FlexVolume:
+- Unnecessary because Rook plans to deprecate FlexVolume support for v1.7
+
+#### `CephObjectStore`
+Dependents which can consume this provider's pools:
+- CephNFS in the same namespace that have `spec.pool == <a provider pool>`
+- CephClient in the same namespace that has any `spec.caps` value with the string `pool=<a provider pool>`
+
+Dependents via lib-bucket-provisioner:
+- ObjectBucketClaims that have:
+  - `spec.endpoint.bucketHost == <provider's service name>.<ceph cluster namespace>.svc` AND
+  - `spec.endpoint.bucketName == <a provider bucket>`
+
+Dependents via COSI:
+- Buckets with a reference to a provider's service and bucket (specific paths TBD by COSI design)
+
+#### `CephObjectZone`, `CephObjectZoneGroup`, and `CephObjectRealm`
+These resources are all part of Rook-Ceph's multi-site object storage strategy.
+
+`CephObjectZone` creates pools. A zone can be effectively thought of as the "object store" itself.
+Dependents which can consume this provider's pools:
+- CephObjectStores in the same namespace that have `spec.zone.name == CephObjectZone.metadata.name`
+- CephNFS in the same namespace that have `spec.pool = <a provider pool>`
+- CephClient in the same namespace that has an `spec.caps` value with the string `pool=<a provider pool>`
+
+`CephObjectRealm` has dependents:
+- CephObjectZoneGroups in the same namespace that have `spec.realm == CephObjectRealm.metadata.name`
+
+`CephObjectZoneGroup` has dependents:
+- CephObjectZones in the same namespace that have `spec.zoneGroup == CephObjectZoneGroup.metadata.name`
+
+#### Mirroring resources
+CephRBDMirror has dependents:
+- CephBlockPools in the same namespace that have `spec.mirroring.enabled == true`
+
+CephFilesystemMirror has dependents:
+- CephFilesystems in the same namespace that have `spec.mirroring.enabled == true`
```

Comment:
I'm not sure what you mean, though you would know best since you are implementing CephFilesystemMirror. From what I can tell, CephFilesystemMirror can't mirror block pools, so I don't think this subsequently follows. It would be good to talk about this specific detail in tomorrow's huddle.

Reply:
Right, CephFilesystemMirror is not mirroring "pool" but it depends on CephFilesystem which depends on CephBlockPool (for fs data and fs metadata).

Reply:
I don't think that's right. The CephFilesystem doesn't require (nor allow) using an existing pool. Therefore, it is cannot be dependent on CephBlockPool. The CephFilesystem controller creates its own pools when it needs them. It still uses the Pool spec, but it is copied and not directed toward a CephBlockPool.

Reply:
Ok, I see you are treating that case differently because `CephFilesystem` only uses the spec and not the CRD.